File: artplayer-kotlin/src/main/java/org/salient/artplayer/MediaStateListener.java
package org.salient.artplayer;

/**
 *  Created by Mai on 2018/7/10
 * *
 *  Description: 视频播放器状态回调
 * *
 */
public interface MediaStateListener {

    // all possible media player internal states
//    ERROR
//    IDLE
//    PREPARING
//    PREPARED
//    PLAYING
//    PAUSED
//    PLAYBACK_COMPLETED

    void onStateError();

    void onStateIdle();

    void onStatePreparing();

    void onStatePrepared();

    void onStatePlaying();

    void onStatePaused();

    void onStatePlaybackCompleted();

    void onSeekComplete();

    void onBufferingUpdate(int progress);

    void onInfo(int what, int extra);

    void onProgressUpdate(int progress, long position, long duration);

    void onEnterSecondScreen();

    void onExitSecondScreen();
}

File: artplayer-kotlin/src/main/java/org/salient/artplayer/MediaStateListener.kt
package org.salient.artplayer

/**
 * Created by Mai on 2018/7/10
 * *
 * Description: 视频播放器状态回调
 * *
 */
interface MediaStateListener {
    // all possible media player internal states
//    ERROR
//    IDLE
//    PREPARING
//    PREPARED
//    PLAYING
//    PAUSED
//    PLAYBACK_COMPLETED
    fun onStateError()

    fun onStateIdle()
    fun onStatePreparing()
    fun onStatePrepared()
    fun onStatePlaying()
    fun onStatePaused()
    fun onStatePlaybackCompleted()
    fun onSeekComplete()
    fun onBufferingUpdate(progress: Int)
    fun onInfo(what: Int, extra: Int)
    fun onProgressUpdate(progress: Int, position: Long, duration: Long)
    fun onEnterSecondScreen()
    fun onExitSecondScreen()
}